File: app/src/main/java/ru/melod1n/schedule/fragment/UpdatesFragment.java
package ru.melod1n.schedule.fragment;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.widget.Toolbar;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.Arrays;

import butterknife.BindView;
import butterknife.ButterKnife;
import ru.melod1n.schedule.MainActivity;
import ru.melod1n.schedule.R;
import ru.melod1n.schedule.adapter.ThemeAdapter;
import ru.melod1n.schedule.items.ThemeItem;

public class UpdatesFragment extends Fragment {

    @BindView(R.id.no_items_container)
    TextView noItems;

    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @BindView(R.id.list)
    RecyclerView list;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_updates, container, false);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        ButterKnife.bind(this, view);

        noItems.setText(R.string.no_updates);

        toolbar.setTitle(R.string.nav_updates);

        DrawerLayout drawerLayout = ((MainActivity) getActivity()).getDrawerLayout();

        ActionBarDrawerToggle toggle = ((MainActivity) getActivity()).initToggle(toolbar);
        drawerLayout.addDrawerListener(toggle);
        toggle.syncState();

        ThemeItem light = new ThemeItem();
        light.setSelected(true);
        light.setName("Light");
        light.setEngineVersion(1);
        light.setMadeBy("μSchedule Dev Team");
        light.setColorSurface("#ffffff");
        light.setColorPrimary("#ffffff");
        light.setColorPrimaryDark("#f8f8f8");
        light.setColorAccent("#FF4081");
        light.setColorBackground("#ffffff");
        light.setColorTabsText("#515151");
        light.setColorControlNormal("#121212");
        light.setColorTextPrimary("#121212");
        light.setColorTextSecondary("#707070");
        light.setColorTextPrimaryInverse("#ffffff");
        light.setColorTextSecondaryInverse("#cccccc");

        ThemeItem dark = new ThemeItem();
        dark.setName("Dark");
        dark.setEngineVersion(1);
        dark.setMadeBy("μSchedule Dev Team");
        dark.setDark(true);
        dark.setColorSurface("#121212");
        dark.setColorPrimary("#161616");
        dark.setColorPrimaryDark("#000000");
        dark.setColorAccent("#7C4DFF");
        dark.setColorBackground("#000000");
        dark.setColorTabsText("#B4B4B4");
        dark.setColorControlNormal("#ffffff");
        dark.setColorTextPrimary("#ffffff");
        dark.setColorTextSecondary("#cccccc");
        dark.setColorTextPrimaryInverse("#121212");
        dark.setColorTextSecondaryInverse("#707070");

        ArrayList<ThemeItem> items = new ArrayList<>(Arrays.asList(light, dark));

        list.setLayoutManager(new LinearLayoutManager(getContext(), RecyclerView.HORIZONTAL, false));

        ThemeAdapter adapter = new ThemeAdapter(this, items);
        list.setAdapter(adapter);

        noItems.setVisibility(View.GONE);

    }
}

File: app/src/main/java/ru/melod1n/schedule/fragment/UpdatesFragment.kt
package ru.melod1n.schedule.fragment

import android.annotation.SuppressLint
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.PopupMenu
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout.OnRefreshListener
import butterknife.ButterKnife
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.no_items.*
import kotlinx.android.synthetic.main.recycler_view.*
import kotlinx.android.synthetic.main.toolbar.*
import ru.melod1n.schedule.R
import ru.melod1n.schedule.activity.MainActivity
import ru.melod1n.schedule.adapter.UpdateAdapter
import ru.melod1n.schedule.base.BaseAdapter
import ru.melod1n.schedule.model.UpdateItem
import java.util.*

class UpdatesFragment : Fragment(), OnRefreshListener, BaseAdapter.OnItemClickListener, BaseAdapter.OnItemLongClickListener {

    private var adapter: UpdateAdapter? = null


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_updates, container, false)
    }

    @SuppressLint("SetTextI18n")
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        ButterKnife.bind(this, view)

        noItemsView.setText(R.string.no_updates)

        toolbar.setTitle(R.string.nav_updates)

        val drawerLayout = (requireActivity() as MainActivity).drawerLayout
        val toggle = (requireActivity() as MainActivity).initToggle(toolbar)

        drawerLayout.addDrawerListener(toggle)
        toggle.syncState()

        noItemsView.visibility = View.GONE
        refreshLayout.setOnRefreshListener(this)

        recyclerView.layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
        prepareData()
    }

    private fun prepareData() {
        val items = arrayListOf<UpdateItem>()

        for (i in 0..19) {
            items.add(UpdateItem("Это хуйня из списка #${i + 1} (Мне надоело писать заглушки)", "https://vk.com/id${i + 1}"))
        }

        createAdapter(items)
    }

    private fun createAdapter(items: ArrayList<UpdateItem>) {
        adapter = UpdateAdapter(requireContext(), items)
        adapter!!.onItemClickListener = this
        adapter!!.onItemLongClickListener = this

        recyclerView.adapter = adapter
        refreshLayout.isRefreshing = false
    }

    override fun onItemClick(position: Int) {
        val url = adapter!!.values[position].url

        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        startActivity(intent)
    }

    override fun onItemLongClick(position: Int) {
        val menu = PopupMenu(requireContext(), requireView())

        menu.gravity = Gravity.TOP or Gravity.END
        menu.inflate(R.menu.fragment_updates_card)
        menu.show()
    }

    override fun onRefresh() {
        prepareData()
    }
}